components/script/dom/htmlvideoelement.rs

```diff
@@ -164,7 +173,16 @@ impl HTMLVideoElement {
         };
 
         // Step 5.
-        self.htmlmediaelement.delay_load_event(true);
+        // This delay must be independent from the ones created by HTMLMediaElement during
+        // its media load algorithm, otherwise a code like
+        // <video poster="poster.png"></video>
+        // (which triggers no media load algorithm unless a explicit call to .load() is done)
+        // will block the document's load event forever.
+        let mut blocker = self.load_blocker.borrow_mut();
```

Comment:
Does this still work correctly if we switch the poster attribute value while the poster is loading? I think we probably need to call LoadBlocker::terminate.

Reply:
Good point. Done.